onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -139,56 +139,79 @@
                 return tensor;
             };
 
-            DmlGraphNodeCreateInfo graphNodeInfo;
+            DmlGraphNodeCreateInfo graphNodeCreateInfo;
             graphNodeProps.internalRegInfo->graphNodeFactoryRegistration->factory(
                 node,
                 constantCpuNodeInputGetter,
                 executionHandle,
-                /*out*/ &graphNodeInfo
+                /*out*/ &graphNodeCreateInfo
             );
-            ORT_THROW_HR_IF(E_UNEXPECTED, !graphNodeInfo.desc);
 
-            uint32_t nodeIndex = gsl::narrow_cast<uint32_t>(graphNodes.size());
-            AbstractOperatorDesc opDesc = *graphNodeInfo.desc; // Make a copy
-
-            // Retrieve lists of input and output tensor descs. These point into the opDesc, which allows us to modify
-            // the tensor descs through these pointers.
-            std::vector<DmlBufferTensorDesc*> inputTensorDescs = opDesc.GetInputTensors();
-            std::vector<DmlBufferTensorDesc*> outputTensorDescs = opDesc.GetOutputTensors();
-
-            // Set connections of the new node
-            for (uint32_t inputIndex = 0; inputIndex < graphNodeInfo.kernelInputIndices.size(); ++inputIndex)
+            // Create a map between operatorGraphNodeIndex to mainGraphNodeIndex.
+            std::unordered_map<uint32_t, uint32_t> operatorGraphNodeIndexToMainGraphNodeIndexMap;
+            uint32_t latestNodeIndex = gsl::narrow_cast<uint32_t>(graphNodes.size());
```

Comment:
>latestNodeIndex [](http://example.com/codeflow?start=21&length=15)

[naming] It appears (reading lines  161 and 169) this isn't the *latest* index, but rather the latest index + 1 index (just past the end). Propose `graphNodeCount`.